torch/csrc/jit/codegen/cuda/fusion.cpp

```diff
@@ -378,7 +389,10 @@ StmtNameType Fusion::registerVal(Val* val) {
 
   val_set_.emplace(val);
   val_deque_.push_back(val);
-  return getValName(*(val->getValType()));
+  const auto vtype = *val->getValType();
+  const auto name = getValName(vtype);
```

Comment:
A quick question just for clarification. When copying from `Fusion` 1 to `Fusion` 2, does the copying of the `Val`s need to proceed in the same order they are registered in `Fusion` 1 so that the name lookup will work?

Reply:
Good question: when cloning a fusion, the names will be copied directly w/o going through this registration path.